src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2326,6 +2330,7 @@ write_deployments_bootswap (OstreeSysroot     *self,
                         cancellable, error))
     return FALSE;
 
+  *out_subbootdir = g_steal_pointer (&new_subbootdir);
```

Comment:
Should you check that `out_subbootdir` isn't `NULL` like in `swap_bootlinks`? It seems like you're updating all the callers of both helpers to pass a valid pointer, but it seems safer in case it's reused.